packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -0,0 +1,572 @@
+   */
+  severity?: (typeof statusSeverityValues)[number];
+  /**
+   * The label to display inside the status
+   */
+  statusLabel?: string;
+  /**
+   * The icon element to display at the end of the Nav Item
+   */
+  endIcon?: ReactElement;
+  /**
+   * Whether the item is expanded by default
+   */
+  isDefaultExpanded?: boolean;
+  /**
+   * Whether the item is expanded
+   */
+  isExpanded?: boolean;
+  /**
+   * Whether the item is disabled
+   */
+  isDisabled?: boolean;
+  /**
+   * Whether the item is active/selected
+   */
+  isSelected?: boolean;
+  /**
+   * Event fired when the nav item is clicked
+   */
+  onClick?(event: MouseEvent<HTMLAnchorElement>): void;
+} & (
+  | {
+      /**
+       * Determines if the side nav item is a section header
+       */
+      isSectionHeader: true;
+      href?: never;
+      children?: never;
+    }
+  | {
+      isSectionHeader?: false;
+      href: string;
+      /**
+       * An array of side nav items
+       */
+      children?: SideNavItem[];
+    }
+);
+
+export type SideNavFooterItem = {
+  id: string;
+  label: string;
+  href: string;
+};
+
+const SideNavCollapsedContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "sideNavCollapsed",
+})(
+  ({
+    odysseyDesignTokens,
+    sideNavCollapsed,
+  }: {
+    odysseyDesignTokens: DesignTokens;
+    sideNavCollapsed: boolean;
+  }) => ({
+    backgroundColor: odysseyDesignTokens.HueNeutral300,
+    paddingTop: odysseyDesignTokens.Spacing5,
+    cursor: "pointer",
+    width: sideNavCollapsed ? "auto" : 0,
+    visibility: sideNavCollapsed ? "visible" : "hidden",
+  }),
+);
+
+const SideNavContainer = styled("div", {
+  shouldForwardProp: (prop) =>
+    prop !== "odysseyDesignTokens" && prop !== "sideNavCollapsed",
+})(
+  ({
+    odysseyDesignTokens,
+    sideNavCollapsed,
+  }: {
+    odysseyDesignTokens: DesignTokens;
+    sideNavCollapsed: boolean;
+  }) => ({
+    backgroundColor: odysseyDesignTokens.HueNeutralWhite,
+    flexDirection: "column",
+    display: "flex",
+    visibility: sideNavCollapsed ? "hidden" : "visible",
+    width: sideNavCollapsed ? "0" : "100%",
+    transitionProperty: "width, visibility",
+    transitionDuration: odysseyDesignTokens.TransitionDurationMain,
+    transitionTimingFunction: odysseyDesignTokens.TransitionTimingMain,
+  }),
+);
+
+const SideNavHeaderContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})(({ odysseyDesignTokens }: { odysseyDesignTokens: DesignTokens }) => ({
+  display: "flex",
+  justifyContent: "space-between",
+  alignItems: "center",
+  paddingLeft: odysseyDesignTokens.Spacing4,
+  paddingRight: odysseyDesignTokens.Spacing4,
+  paddingTop: odysseyDesignTokens.Spacing3,
+  paddingBottom: odysseyDesignTokens.Spacing3,
+}));
+
+const CollapseIcon = ({ onClick }: { onClick?(): void }): ReactElement => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+  return (
+    <div
```

Comment:
IMO, we should stick with `<Box>` instead of `<div>` for consistency. Curious what @KevinGhadyani-Okta and @bryancunningham-okta think.

Reply:
I'd think this is better off being an actual `button` personally. If we are going to keep it as a `div`, we need to handle the Space key press as well

> Note: If using role="button" instead of the semantic <button> or <input type="button"> elements, you will need to make the element focusable and define event handlers for [click](https://developer.mozilla.org/en-US/docs/Web/API/Element/click_event) and [keydown](https://developer.mozilla.org/en-US/docs/Web/API/Element/keydown_event) events. This includes handling the Enter and Space keypresses in order to process all forms of user input. See [the official WAI-ARIA example code](https://www.w3.org/WAI/ARIA/apg/patterns/button/examples/button/).

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role

Reply:
that's a great suggestion. I did try the Box/Button option but had to override the button styles to make it look like the icon in the design. So going with the button option esp as it handles all the keyboard events as Bryan mentioned.

Reply:
@ganeshsomasundaram-okta Which icon/button in the design is this? This one? 
<img width="51" alt="image" src="https://github.com/okta/odyssey/assets/148223016/d491f3b2-6324-4e6b-883f-c3cd2bfa05ee">

Reply:
yes, I found a `CollapseLeftIcon` in our icons list and using that inside the Button now.